src/wrapped_radix_engine_toolkit/low_level/transaction_manifest/transaction_manifest.rs

```diff
@@ -0,0 +1,339 @@
+        blobs: Blobs,
+    ) -> Self {
+        Self {
+            secret_magic: TransactionManifestSecretMagic::new(
+                instructions,
+                blobs,
+            ),
+        }
+    }
+}
+
+impl TransactionManifest {
+    pub(crate) fn empty(network_id: NetworkID) -> Self {
+        Self {
+            secret_magic: TransactionManifestSecretMagic {
+                instructions: Instructions::empty(network_id),
+                blobs: Blobs::default(),
+            },
+        }
+    }
+}
+
+impl From<TransactionManifestSecretMagic> for TransactionManifest {
+    fn from(value: TransactionManifestSecretMagic) -> Self {
+        Self {
+            secret_magic: value,
+        }
+    }
+}
+
+impl TransactionManifest {
+    pub(crate) fn scrypto_manifest(&self) -> ScryptoTransactionManifest {
+        ScryptoTransactionManifest {
+            instructions: self.instructions().clone(),
+            blobs: self.secret_magic.blobs.clone().into(),
+        }
+    }
+}
+
+impl From<TransactionManifest> for ScryptoTransactionManifest {
+    fn from(value: TransactionManifest) -> Self {
+        value.scrypto_manifest()
+    }
+}
+
+impl TransactionManifest {
+    pub(crate) fn instructions(&self) -> &Vec<ScryptoInstruction> {
+        &self.secret_magic.instructions.secret_magic.0
+    }
+
+    pub(crate) fn blobs(&self) -> &Blobs {
+        &self.secret_magic.blobs
+    }
+
+    pub(crate) fn from_scrypto(
+        scrypto_manifest: ScryptoTransactionManifest,
+        network_id: NetworkID,
+    ) -> Self {
+        let value = Self {
+            secret_magic: TransactionManifestSecretMagic {
+                instructions: Instructions {
+                    secret_magic: InstructionsSecretMagic(
+                        scrypto_manifest.instructions.clone(),
+                    ),
+                    network_id,
+                },
+                blobs: scrypto_manifest.blobs.clone().into(),
+            },
+        };
+        assert_eq!(value.scrypto_manifest(), scrypto_manifest);
+        value
+    }
+
+    pub fn new(
+        instructions_string: impl AsRef<str>,
+        network_id: NetworkID,
+        blobs: Blobs,
+    ) -> Result<Self> {
+        Instructions::new(instructions_string, network_id).map(|instructions| {
+            Self {
+                secret_magic: TransactionManifestSecretMagic {
+                    instructions,
+                    blobs,
+                },
+            }
+        })
+    }
+
+    pub fn instructions_string(&self) -> String {
+        self.secret_magic.instructions.instructions_string()
+    }
+
+    pub fn summary(&self) -> ManifestSummary {
+        let ret_summary = RET_summary(&self.scrypto_manifest());
+        ManifestSummary::from((ret_summary, self.network_id()))
+    }
+
+    pub fn network_id(&self) -> NetworkID {
+        self.secret_magic.instructions.network_id
+    }
+
+    pub fn involved_resource_addresses(&self) -> Vec<ResourceAddress> {
```

Comment:
`RET_ins_extract_addresses` is function: `radix_engine_toolkit::functions::instructions::extract_addresses` which is NOT throwing, so this method need not be throwing